src/plugins/ui_actions/public/context_menu/build_eui_context_menu_panels.test.ts

```diff
@@ -36,46 +36,335 @@ const createTestAction = ({
+        trigger: 'TETS_TRIGGER' as any,
+      },
+    ],
+    closeMenu: () => {},
+  });
+
+  expect(menu2[0].items![0].name).toBe('Bar');
+  expect(menu2[0].items![1].name).toBe('Foo');
+});
+
+test('hides items behind in "More" submenu if there are more than 4 actions', async () => {
+  const menu = await buildContextMenuForActions({
+    actions: [
+      {
+        action: {
+          id: 'foo1',
+          type: 'foo' as any,
+          order: 1,
+          getDisplayName: () => 'Foo 1',
+          getIconType: () => 'empty',
+          isCompatible: async () => true,
+          execute: async () => {},
+        },
+        context: {},
+        trigger: 'TETS_TRIGGER' as any,
+      },
+      {
+        action: {
+          id: 'foo2',
+          type: 'foo' as any,
+          order: 1,
+          getDisplayName: () => 'Foo 2',
+          getIconType: () => 'empty',
+          isCompatible: async () => true,
+          execute: async () => {},
+        },
+        context: {},
+        trigger: 'TETS_TRIGGER' as any,
+      },
+      {
+        action: {
+          id: 'foo3',
+          type: 'foo' as any,
+          order: 1,
+          getDisplayName: () => 'Foo 3',
+          getIconType: () => 'empty',
+          isCompatible: async () => true,
+          execute: async () => {},
+        },
+        context: {},
+        trigger: 'TETS_TRIGGER' as any,
+      },
+      {
+        action: {
+          id: 'foo4',
+          type: 'foo' as any,
+          order: 1,
+          getDisplayName: () => 'Foo 4',
+          getIconType: () => 'empty',
+          isCompatible: async () => true,
+          execute: async () => {},
+        },
+        context: {},
+        trigger: 'TETS_TRIGGER' as any,
+      },
+      {
+        action: {
+          id: 'foo5',
+          type: 'foo' as any,
+          order: 1,
+          getDisplayName: () => 'Foo 5',
+          getIconType: () => 'empty',
+          isCompatible: async () => true,
+          execute: async () => {},
+        },
+        context: {},
+        trigger: 'TETS_TRIGGER' as any,
+      },
+    ],
+    closeMenu: () => {},
+  });
+
+  expect(menu).toMatchInlineSnapshot(`
+    Array [
+      Object {
+        "id": "mainMenu",
+        "items": Array [
+          Object {
+            "data-test-subj": "embeddablePanelAction-foo1",
+            "href": undefined,
+            "icon": "empty",
+            "name": "Foo 1",
+            "onClick": [Function],
+            "panel": undefined,
+          },
+          Object {
+            "data-test-subj": "embeddablePanelAction-foo2",
+            "href": undefined,
+            "icon": "empty",
+            "name": "Foo 2",
+            "onClick": [Function],
+            "panel": undefined,
+          },
+          Object {
+            "data-test-subj": "embeddablePanelAction-foo3",
+            "href": undefined,
+            "icon": "empty",
+            "name": "Foo 3",
+            "onClick": [Function],
```

Comment:
This snapshot tests are excessive and will have to be updated if EUI adds new required property in the panel. Since we want to test menu structure, I'd suggest to first map the menu to have only panel names, and then `toMatchInlineSnapshot` structure of names. 

It also would be clearer to review this test's results from a glance.

Reply:
Done.